onnxruntime/core/graph/contrib_ops/quantization_defs.cc

```diff
@@ -1132,88 +1132,225 @@ If mask is provided, mask index (that is position of first 0 in mask, or number
-                                    propagateElemTypeFromInputToOutput(ctx, 0, 0);
-                                  }));
+  ONNX_MS_OPERATOR_SET_SCHEMA(
+      QOrderedMatMul,
+      1,
+      OpSchema()
+          .SetDoc(QOrderedMatMul_ver1_doc)
+          .Attr("order_A", "cublasLt order of matrix A", AttributeProto::INT)
+          .Attr("order_B", "cublasLt order of matrix B", AttributeProto::INT)
+          .Attr("order_Y", "cublasLt order of matrix Y and optional matrix C", AttributeProto::INT)
+          .Input(0, "A", "3-dimensional matrix A", "Q")
+          .Input(1, "scale_A", "scale of the input A", "S")
+          .Input(2, "B", "2-dimensional matrix B", "Q")
+          .Input(3, "scale_B", "scale of the input B. Scalar or 1-D float32", "S")
+          .Input(4, "scale_Y", "scale of the output Y", "S")
+          .Input(5, "bias", "1d bias", "S", OpSchema::Optional)
+          .Input(6, "C", "3d or 2d matrix C. if 2d expand to 3d first. Shape[0] should be 1 or same as A.shape[0] ", "Q", OpSchema::Optional)
+          .Input(7, "scale_C", "scale of the input A", "S", OpSchema::Optional)
+          .Output(0, "Y", "Matrix multiply results from A * B", "Q")
+          .TypeConstraint("Q", {"tensor(int8)"}, "Constrain input and output types to int8 tensors.")
+          .TypeConstraint("S", {"tensor(float)"}, "Constrain bias and scales to float32")
+          .TypeAndShapeInferenceFunction([](ONNX_NAMESPACE::InferenceContext& ctx) {
+            propagateElemTypeFromInputToOutput(ctx, 0, 0);
+            ONNX_NAMESPACE::matmulShapeInference(ctx, 0, 2);
+          }));
+
+  static const char* Attention_QOrdered_doc = R"DOC(
+Quantized version of simplified Multi-Head Self Attention(using int8 with specific matrix Layout).
+Multi-Head Self Attention that can be either unidirectional (like GPT-2) or bidirectional (like BERT).
+The mask_index input is optional. Besides raw attention mask with shape (batch_size, past_sequence_length + sequence_length)
+or (batch_size, sequence_length, past_sequence_length + sequence_length) with value 0 for masked and 1 otherwise,
+we also support other two formats: When input has right-side padding, mask_index is one dimension with shape (batch_size),
+where value of each element is the end position, or valid length of actual sequence excluding padding. When input has
+left-side padding, mask_index has shape (2 * batch_size), where the values are the exclusive end positions followed by
+the inclusive start positions. When unidirectional is 1, and each token only attend to previous tokens. For GPT-2, both past
+and present state are optional. Present state could appear in output even when past state is not in input.
+Current version does not support past/present, extra_add and qkv_hidden_sizes.
+TODO: Support them if needed in the future.
+)DOC";
 
-  ONNX_MS_OPERATOR_SET_SCHEMA(QOrderedGelu, 1,
-                              OpSchema()
```

Comment:
It seems like the existing schemas for QOrderedMatMul, QOrderedLayerNormalization, QOrderedGelu have been deleted ?

Reply:
Not removed, they are still there. Updated MatMul. too many commit cause confusion.